libsolidity/interface/CompilerStack.cpp

```diff
@@ -260,8 +260,19 @@ bool CompilerStack::analyze()
 	{
 		SyntaxChecker syntaxChecker(m_errorReporter, m_optimiserSettings.runYulOptimiser);
 		for (Source const* source: m_sourceOrder)
-			if (!syntaxChecker.checkSyntax(*source->ast))
+		{
+			if (source->ast)
```

Comment:
Couldn't this already be `if (source->ast && !syntaxChecker.checkSyntax(*source->ast))`? You can also leave out the curly brackets for `if`-Statements, that have only one statement in their body ;)

Reply:
@erak 

```
for (Source const* source: m_sourceOrder)
    if (!syntaxChecker.checkSyntax(*source->ast) && if (source->ast)
        noErrors = false;
```

is semantically not the same as what I have, but then I could have misunderstood exactly what change you meant. 

I have no intrinsic interest in changes of this ilk; I am more focused on user-facing problems like whether we in running `solc` there is a SEGV or not, or that IDE users can get AST's when requested in the presence of errors. 

Of course I understand matters of style and how various boolean expressions are expressed seem to be of concern to you and possibly others and it is not my intention to disuade you and others from such activity. I just ask that as a _volunteer_ here you not involve me since I have no interest in this. 

So I suggest that for such kinds of things, you go ahead and just make the changes, _compile_ the result and run the tests. If the tests check out, I am perfectly happy to go with whichever form is most pleasing to you.